subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerDaemonIntegrationTest.groovy

```diff
@@ -126,6 +126,7 @@ class WorkerDaemonIntegrationTest extends AbstractWorkerExecutorIntegrationTest
             task runInDaemon(type: WorkerTask) {
                 isolationMode = 'processIsolation'
                 workActionClass = ${workActionThatVerifiesOptions.name}.class
+                def fileTree = (${isOracleJDK}) ? project.fileTree(new File(Jvm.current().jre, "lib")).include("*.jar") : null
```

Comment:
Did you find out why the original code did not require this condition?

Reply:
This conditional assignment was in a previous version of the code but was made unconditional during some refactoring [here](https://github.com/gradle/gradle/commit/3d06528e6994b508a0300f0b873f6da89496fa14#diff-5eed20fa42d42099f10e41681d22be1d7827a66b09969f83e903871ba4088925L143). I guess enabling load-after-store surfaced the issue, but I'm not 100% sure why.